Release v0.51.0-beta.0 (#2737)
perses/perses: CHANGELOG.md

# Changelog

## 0.51.0-beta.0 / 2025-03-10

This release is one of the biggest releases we ever made. 
On top of many features and improvements, we are introducing a new plugin architecture that allows you to develop external plugins and load them in Perses. 

> As part of this change, we are also introducing a new repository `perses/plugins` containing all default plugin supported by Perses.

Reviewer: ```suggestion
As part of this change we are introducing a new repository `perses/plugins` containing all default plugins supported by Perses.
```

> We have also improved the documentation, highlighting features that were implemented one or two years ago without being used from what we heard (like the discovery or the ephemeral dashboard.)

Reviewer: ```suggestion
We have also improved the documentation, highlighting features that were implemented some time ago already but not well advertised (like datasource discovery and ephemeral dashboards)
```

- [BUGFIX] ListVariable: do not clear input value when selecting a value (#2672)
- [BUGFIX] Refresh RBAC cache when adding or removing users. (#2670)
- [BUGFIX] Dashboard copy: hide ephemeral option if disabled (#2633)
- [BUGFIX] Dashboard edit mode - Expanded panel - delete & drag (#2625)
- [BUGFIX] CUE lib: Add missing model files (#2612)
- [BUGFIX] Uses pkg filepath instead of path when manipulating file access (#2611)
- [BUGFIX] Wait until plugins are loaded to avoid invalid default queries for explore components (#2607)
- [BUGFIX] Add id to avoid setState loops (#2600)
- [BUGFIX] Inject API prefix in all UI assets (#2576)
- [BUGFIX] Login from percli (method not allowed) (#2577)
- [BUGFIX] ListVariable: fix autocomplete when 'All' is selected (#2554)
- [BUGFIX] Update unit's migration (#2519)
- [BUGFIX] GaugeChartPanel: fix empty state (#2548)
- [BUGFIX] Fixes tooltip display in EDGE / IE (#2547)
- [BUGFIX] Redirect to the desired page after login success (#2354)
- [BUGFIX] DaC CUE SDK: Fix inconsistencies in mandatory vs optional attributes in dashboard lib (#2540)
- [BUGFIX] `percli dac setup`: fix useless requirement of go CLI when using CUE (#2544)

> - [BREAKINGCHANGE] Changes the plugin architure to support remote plugins. (#2154) (#2511) (#2526) (#2545) (#2567) (#2578) (#2582) (#2593) (#2598) (#2606) (#2609) (#2613) (#2616) (#2626) (#2637) (#2660) (#2645) (#2690) (#2719)

Reviewer: ```suggestion
- [BREAKINGCHANGE] Changes the plugin architecture to support remote plugins. (#2154) (#2511) (#2526) (#2545) (#2567) (#2578) (#2582) (#2593) (#2598) (#2606) (#2609) (#2613) (#2616) (#2626) (#2637) (#2660) (#2645) (#2690) (#2719)
```

Reviewer: I would also move #2695 in this list, WDYT?

Author: ah yeah good catch, for both notes
